src/native/libs/System.Native/pal_io.c

```diff
@@ -1638,6 +1638,19 @@ int32_t SystemNative_LChflags(const char* path, uint32_t flags)
 #endif
 }
 
+int32_t SystemNative_FChflags(intptr_t fd, uint32_t flags)
+{
+#if HAVE_LCHFLAGS
+    int32_t result;
+    while ((result = fchflags(ToFileDescriptor(fd), flags)) < 0 && errno == EINTR);
```

Comment:
Not suggesting this is wrong (and it follows pattern above) but the second parameter is defined to take `unsigned long`, which seems to vary in size:
```
dan@danmoseL:/usr/include$ grep -C3 ULONG_MAX *h
limits.h-
limits.h-/* Maximum value an `unsigned long int' can hold.  (Minimum is 0.)  */
limits.h-#  if __WORDSIZE == 64
limits.h:#   define ULONG_MAX   18446744073709551615UL
limits.h-#  else
limits.h:#   define ULONG_MAX   4294967295UL
limits.h-#  endif
```
whereas uint32_t being passed here is always 32 bits. Curious what I'm missing @tmds

Reply:
Yes, a long can be 64-bit.

There are some other strange things about this.

UF_HIDDEN is filtered and represented with a PAL constant:

https://github.com/dotnet/runtime/blob/5992145db2cb57956ee444aa0f0c2f3f85ee3673/src/native/libs/System.Native/pal_io.c#L219

So the other flags are lost.

And when setting, the PAL constant is not mapped back to the platform constant.

Reply:
This was added in https://github.com/dotnet/corefx/pull/34560.

Reply:
Interesting. Well, it might be best to leave this as is in this PR, as it clearly "works" in some sense following SystemNative_LChflags, and open an issue to clean this up.

Reply:
I would mark this as "resolved".
@tmds @danmoseley Could one of you then create the issue so that this comment is not forgotten, since I am not so experienced in the Unix file system topic and you have more expertise in the area.

Reply:
See #69005